config/dynamic.c

```diff
@@ -16,11 +16,527 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <sys/un.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/select.h>
+#include <lualib.h>
+#include <lauxlib.h>
+
+#include <rte_log.h>
+#include <rte_lcore.h>
+#include <rte_cycles.h>
+#include <rte_malloc.h>
+
+#include "gatekeeper_net.h"
+#include "gatekeeper_main.h"
 #include "gatekeeper_config.h"
+#include "gatekeeper_launch.h"
+
+/* TODO Get the install-path via Makefile. */
+#define LUA_DY_BASE_DIR     "./lua"
+#define DYNAMIC_CONFIG_FILE "dy.lua"
+
+const char *CLIENT_PROC_ERROR =
+	"function reply() return -1, \"Dynamic Configuration cannot process the request. end\"\n";
+
+const uint16_t MSG_MAX_LEN = (uint16_t)~0U;
+
+/*
+ * Return values:
+ * 	0: Write @nbytes successfully.
+ * 	-1: Connection closed by the client.
+ */
+static int
+write_nbytes(int conn_fd, const char *msg_buff, int nbytes)
+{
+	int send_size;
+	int tot_size = 0;
+
+	while ((send_size = write(conn_fd, msg_buff + tot_size,
+			nbytes - tot_size)) > 0) {
+		tot_size += send_size;
+		if (tot_size >= nbytes)
+			break;
+	}
+
+	/* The connection with the client is closed. */
+	if (send_size == 0) {
+		RTE_LOG(WARNING, GATEKEEPER,
+			"dyn_cfg: Client disconnected\n");
+		return -1;
+	} else if (send_size < 0) {
```

Comment:
Style tip: you don't need the `else` since the previous `if` finishes with a `return`. This make the code just a sequence of `if` statement, what is easier to read.